### PR TITLE
Issue #9941: Fixing AtclauseOrder falsely ignoring method with annotation

### LINKED ISSUE #9941: AtclauseOrder: Falsely ignores method with annotation
Identified at https://github.com/checkstyle/checkstyle/pull/9939

````
$ cat TestClass.java
public class TestClass {
    /**
     * Test.
     *
     * @deprecated Test.
     * @param type Test.
     * @return Test.
     */
    @Deprecated
    public boolean branchContains(int type) { return true; }

    /**
     * Test.
     *
     * @deprecated Test.
     * @param type Test.
     * @return Test.
     */
    public boolean branchContains2(int type) { return true; }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="AtclauseOrder">
      <property name="tagOrder" value="@param, @return, @throws, @deprecated"/>
      <property name="target"
               value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-8.41.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:16: Block tags have to appear in the order '[@param, @return, @throws, @deprecated]'. [AtclauseOrder]
[ERROR] TestClass.java:17: Block tags have to appear in the order '[@param, @return, @throws, @deprecated]'. [AtclauseOrder]
Audit done.
Checkstyle ends with 2 errors.
````

Method without annotation prints 2 violations while one with annotation prints none.

Method annotations should not play into this check's behavior and I expect the same violations between both examples.

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #9941: Fixing AtclauseOrder falsely ignoring method with annotation
Diff Regression projects: https://gist.githubusercontent.com/Vyom-Yadav/91807e6244cff60698d9e33e32ba2d51/raw/eb228fa41edadc10eea264a914ea1411f0e52122/projects-to-test-on.properties
Diff Regression config: https://gist.githubusercontent.com/Vyom-Yadav/98dceb63a79f4833e85fff9b2e1464a6/raw/4e042b7368c2ab4a66635033e0cb338c81f1d7f8/my_checks.xml

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
Github, generate report

**Comment by Vyom-Yadav** [comment]
Github, generate report

**Comment by Vyom-Yadav** [comment]
Report looks fine to me.

**Comment by nmancus1** [comment]
wercker is failing on ./.ci/wercker.sh no-error-pgjdbc .

<details>

```
export WERCKER_STEP_ROOT="/pipeline/script-a4cae8e5-99e6-4197-8761-5b827d19989d"
export WERCKER_STEP_ID="script-a4cae8e5-99e6-4197-8761-5b827d19989d"
export WERCKER_STEP_OWNER="wercker"
export WERCKER_STEP_NAME="script"
export WERCKER_REPORT_NUMBERS_FILE="/report/script-a4cae8e5-99e6-4197-8761-5b827d19989d/numbers.ini"
export WERCKER_REPORT_MESSAGE_FILE="/report/script-a4cae8e5-99e6-4197-8761-5b827d19989d/message.txt"
export WERCKER_REPORT_ARTIFACTS_DIR="/report/script-a4cae8e5-99e6-4197-8761-5b827d19989d/artifacts"
source "/pipeline/script-a4cae8e5-99e6-4197-8761-5b827d19989d/run.sh" < /dev/null
Command: ./.ci/wercker.sh no-error-pgjdbc
CS_version: 8.43-SNAPSHOT
Cloning into 'pgjdbc'...
Note: checking out 'bad34f1c2985225b8809eb1c3fee5c5684d363a7'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at bad34f1c Correct code style for pgjdbc according to new checkstyle changes
Downloading https://services.gradle.org/distributions/gradle-6.7.1-bin.zip
.........10%..........20%..........30%..........40%..........50%.........60%..........70%..........80%..........90%..........100%

Welcome to Gradle 6.7.1!

Here are the highlights of this release:
 - File system watching is ready for production use
 - Declare the version of Java your build requires
 - Java 15 support

For more details see https://docs.gradle.org/6.7.1/release-notes.html

To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/6.7.1/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing
> Task :buildSrc:pluginDescriptors
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:autostyleKotlinCheck
> Task :buildSrc:autostyleKotlinGradleCheck
> Task :buildSrc:autostyleCheck
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:compileKotlin
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC

> Task :buildSrc:jar
:jar: No valid plugin descriptors were found in META-INF/gradle-plugins

> Task :buildSrc:assemble
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:validatePlugins FROM-CACHE
> Task :buildSrc:check
> Task :buildSrc:build

> Configure project :
Building pgjdbc 42.3.0-SNAPSHOT

> Task :checkstyleAll UP-TO-DATE
> Task :benchmarks:compileJava NO-SOURCE
> Task :benchmarks:processResources NO-SOURCE
> Task :benchmarks:classes UP-TO-DATE
> Task :benchmarks:compileTestJava NO-SOURCE
> Task :benchmarks:processTestResources NO-SOURCE
> Task :benchmarks:testClasses UP-TO-DATE
> Task :postgresql:preprocessVersion

> Task :postgresql:compileJava
/pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/util/StreamWrapper.java:120: warning: [removal] finalize() in FileInputStream has been deprecated and marked for removal
          protected void finalize() throws IOException {
                         ^
/pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/util/StreamWrapper.java:126: warning: [removal] finalize() in FileInputStream has been deprecated and marked for removal
              super.finalize();
                   ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 warnings

> Task :benchmarks:compileJmhJava
> Task :benchmarks:processJmhResources NO-SOURCE
> Task :benchmarks:jmhClasses
> Task :benchmarks:checkstyleJmh
> Task :benchmarks:checkstyleMain NO-SOURCE
> Task :benchmarks:checkstyleTest NO-SOURCE
> Task :benchmarks:checkstyleAll
> Task :bom:checkstyleAll UP-TO-DATE

[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:42: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]> Task :postgresql:checkstyleMain

[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:43: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:44: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:72: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:73: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:74: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]
[ant:checkstyle] [ERROR] /pipeline/source/.ci-temp/pgjdbc/pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java:75: Block tags have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [AtclauseOrder]

> Task :postgresql:checkstyleMain FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':postgresql:checkstyleMain'.
> Checkstyle rule violations were found. See the report at: file:///pipeline/source/.ci-temp/pgjdbc/pgjdbc/build/reports/checkstyle/main.html
  Checkstyle files with violations: 1
  Checkstyle violations by severity: [error:7]


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.


* Get more help at https://help.gradle.org

BUILD FAILED in 3m 37s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings
5 actionable tasks: 5 executed
S3 cache 112ms wasted on misses, reads: 23, hits: 1, elapsed: 1055ms, processed: 406 B
```

</details>

**Comment by Vyom-Yadav** [comment]
@nmancus1 One more thing, while running the command `./.ci/wercker.sh no-error-pgjdbc` the build could not complete due to-
```
* What went wrong:
Execution failed for task ':benchmarks:checkstyleJmh'.
> Could not resolve all files for configuration ':benchmarks:checkstyle'.
   > Could not find com.puppycrawl.tools:checkstyle:8.43-SNAPSHOT.
```
However I got the missing snapshot by running the command `mvn -e clean install -Pno-validations` and then the build completed. I encountered this error earlier also and @rnveach gave me the answer in https://github.com/checkstyle/checkstyle/pull/9334#issuecomment-787517588 . Like rnveach said I think this should run without manual intervention, can you open an issue for it as I don't know much about scripting.

**Comment by nmancus1** [comment]
> However I got the missing snapshot by running...

This is intentional decision from https://github.com/checkstyle/checkstyle/pull/8950

**Comment by Vyom-Yadav** [comment]
@nmancus1 I read the comments in #8950 and they all seemed justified.  I will start learning shell scripting from now onward, but I think that this should be documented. Also isn't there any way to do something like-
```
if (!snapshot_is_present) {
mvn -e --no-transfer-progress clean install -Pno-validations
}
```
for all the scripts that use the snapshot?

**Comment by nmancus1** [comment]
> Also isn't there any way to do something like ...

This is a good comment to place in the issue that you create :)

**Comment by Vyom-Yadav** [comment]
@rnveach Can you review?

**Review by rnveach — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Comment by Vyom-Yadav** [comment]
@romani Kindly complete review

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheckTest.java [hunk 1]
```diff
@@ -173,4 +173,36 @@ public void testAtclauseOrderLotsOfRecords() throws Exception {
         verify(checkConfig, getNonCompilablePath("InputAtclauseOrderLotsOfRecords.java"), expected);
     }
 
+    @Test
+    public void testAtclause() throws Exception {
+        final String tagOrder = "[@author, @version, @param, @return, @throws, @exception, @see,"
+                + " @since, @serial, @serialField, @serialData, @deprecated]";
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheckTest.java:179]: this is not default compare with https://checkstyle.org/config_javadoc.html#AtclauseOrder_Properties

message suppose to print expected order:
https://github.com/checkstyle/checkstyle/blob/bf86a3ad249ce60fdf90a433d5e9f739e56a9cca/src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages.properties#L1

please recheck what is root of problem, there might be outdated value in html, please investigate and lets create  separate issue on this.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheckTest.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheckTest.java:194]: Issue created in #10086

**Pull request merged**